### PR TITLE
Implementation of Hover Feature

### PR DESCRIPTION
**Pull request by Jg1255**

### CODE REVIEW

File: packages/compiler/server/serverlib.ts [hunk 1]
```diff
@@ -606,6 +613,31 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function getDocument(params: DefinitionParams): Promise<string> {
+    const sym = await compile(params.textDocument, (program, document, file) => {
```
Comment (timotheeguerin) [inline packages/compiler/server/serverlib.ts:617]: Here you should really look into getting info from the `Type` not a `Symbol`. There a a `type` property in the sym.

Comment (timotheeguerin) [inline packages/compiler/server/serverlib.ts:617]: That compile should also go in `getHoverSymbol` and this funciton should take in a `Type`

Comment (timotheeguerin) [reply packages/compiler/server/serverlib.ts:617]: it should really be `getTypeDetails(type: Type): string`

File: packages/compiler/server/serverlib.ts [hunk 2]
```diff
@@ -606,6 +613,31 @@ export function createServer(host: ServerHost): Server {
...
+    if (symbol) {
+      return [symbol.name, symbol.declarations[0].kind.toString()].join(", ");
+    }
+    return "";
+  }
+
+  async function getHoverSymbol(params: HoverParams): Promise<Hover> {
+    const docString = await getDocument(params);
+    const markdown: MarkupContent = {
+      kind: MarkupKind.PlainText,
```
Comment (timotheeguerin) [inline packages/compiler/server/serverlib.ts:631]: we might want to set markdown here and be able to have a better detail

File: packages/compiler/server/server.ts [hunk 3]
```diff
@@ -73,6 +73,8 @@ function main() {
   connection.onFoldingRanges(s.getFoldingRanges);
   connection.onDocumentSymbol(s.getDocumentSymbols);
   connection.onDocumentHighlight(s.findDocumentHighlight);
+  connection.onHover(s.getHoverSymbol);
```
Comment (timotheeguerin) [inline packages/compiler/server/server.ts:76]: Do the same change in the playground

File: packages/compiler/server/server.ts [hunk 4]
```diff
@@ -73,6 +73,8 @@ function main() {
...
+
```
Comment (timotheeguerin) [inline packages/compiler/server/server.ts]: Also should be able to add the detail information to the `CompleteList`

File: packages/compiler/server/serverlib.ts (hunk 1 above)
Comment (nguerrera) [reply packages/compiler/server/serverlib.ts:617]: Careful, Sym.type is only populated for late-bound symbols, I think. addIdentifierCompletion does this:

image

getTypeForSymbol might be a good helper to have for this.

**Pull request closed without merging**